### DISCUSSION TITLE
install cannot find built package in gh action macos worker

### DISCUSSION THREAD

**Discussion by Maxcode123** [pypa/setuptools#4677] [Q&A] [answered]
I'm writing a C extension (called [dictionaries](https://github.com/Maxcode123/dictionaries/tree/3-ssdict-mapping-methods)) and when building the package with `python setup.py install` on github-actions workers everything looks fine for linux and windows (who knew?) workers whereas for macos workers after successfully installing the package it goes on to look for it on pypi.

Here are the [logs](https://github.com/Maxcode123/dictionaries/actions/runs/11294283119/job/31414352415) from the macos-worker:
```
Installed /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/dictionaries-0.1.0-py3.10-macosx-10.9-universal2.egg
Processing dependencies for dictionaries==0.1.0
Searching for dictionaries==0.1.0
Reading https://pypi.org/simple/dictionaries/
/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning:  is an invalid version and will not be supported in a future release
  warnings.warn(
No local packages or working download links found for dictionaries==0.1.0
error: Could not find suitable distribution for Requirement.parse('dictionaries==0.1.0')
```

Linux worker [logs](https://github.com/Maxcode123/dictionaries/actions/runs/11294530227/job/31415181877):
```
Installed /opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/dictionaries-0.1.0-py3.10-linux-x86_64.egg
Processing dependencies for dictionaries==0.1.0
Finished processing dependencies for dictionaries==0.1.0
```
It seems like on Linux and Windows when processing dependencies, after having installed the package, setuptools find the local package, but on MacOs it doesn't.

Is this some known behaviour of MacOS systems?

**Comment by abravalheri** [comment, marked as answer]
Please don't use `python setup.py install`. Using `setup.py` as a CLI tool is deprecated (`setup.py` should only be used as a configuration file, not CLI).

Does the error persist if you use `pip` instead?

**Reply by Maxcode123** [reply]
Hey, thanks for the answer.
I'm trying to build the package and that's where the error originates; I'm not sure how to build the package using `pip`, maybe you mean I should use `python -m build`?

**Reply by Maxcode123** [reply]
Scratch the above; I think I got it now. I used `pip install -e .` to install the local package. Thanks for the support :)
